src/urllib3/util/retry.py

```diff
@@ -27,6 +28,49 @@
 )
 
 
+# TODO: In v2 we can remove this sentinel and metaclass with deprecated options.
+_Default = object()
+
+
+class RetryMeta(type):
```

Comment:
Optional nit: Maybe `RetryDeprecations` or `RetryWarnings` would be more descriptive.

Reply:
I'm going to pass on this one since it's "Retry metaclass", maybe I should make the class private though.

Reply:
Maybe I just haven't written Python in too long, it has become weird to me to add type hinting into the name of things (rather than describe what they do).